### DISCUSSION TITLE
Caddy stuck at "trying to solve challenge"

### DISCUSSION THREAD

**Discussion by fabifont** [coder/code-server#4051] [Q&A]
## OS/Web Information

- Web Browser: Every browser
- Local OS: Linux arch 5.13.12-arch1-1 #1 SMP PREEMPT Wed, 18 Aug 2021 20:49:03 +0000 x86_64 GNU/Linux
- Remote OS: Linux arch 5.13.12-arch1-1 #1 SMP PREEMPT Wed, 18 Aug 2021 20:49:03 +0000 x86_64 GNU/Linux
- Remote Architecture: x86
- `code-server --version`: [Latest docker code-server](https://hub.docker.com/layers/linuxserver/code-server/latest/images/sha256-967c9f521d3f08f0ab182c233b7e910d230a54910dcbf81c24364974c26f4db0?context=explore)

## Steps to Reproduce
I made a repository one year ago to use code-server and caddy both dockerized and it was working. After a year I tried to run the same setup but it didn't work anymore (stuck with caddy trying to solve challenge). Here you can find my repo: https://github.com/fabifont/caddy-code-server . You can follow the README to test it.

## Expected
Everything works.

## Actual
Seems like Caddy doesn't work.

## Logs
code-server    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
code-server    | [s6-init] ensuring user provided files have correct perms...exited 0.
code-server    | [fix-attrs.d] applying ownership & permissions fixes...
code-server    | [fix-attrs.d] done.
code-server    | [cont-init.d] executing container initialization scripts...
code-server    | [cont-init.d] 01-envfile: executing...
code-server    | [cont-init.d] 01-envfile: exited 0.
code-server    | [cont-init.d] 10-adduser: executing...
code-server    | usermod: no changes
code-server    |
code-server    | -------------------------------------
code-server    |           _         ()
code-server    |          | |  ___   _    __
code-server    |          | | / __| | |  /  \
code-server    |          | | \__ \ | | | () |
code-server    |          |_| |___/ |_|  \__/
code-server    |
code-server    |
code-server    | Brought to you by linuxserver.io
code-server    | -------------------------------------
code-server    |
code-server    | To support LSIO projects visit:
code-server    | https://www.linuxserver.io/donate/
code-server    | -------------------------------------
code-server    | GID/UID
code-server    | -------------------------------------
code-server    |
code-server    | User uid:    1001
code-server    | User gid:    1001
code-server    | -------------------------------------
code-server    |
code-server    | [cont-init.d] 10-adduser: exited 0.
code-server    | [cont-init.d] 30-config: executing...
code-server    | setting up sudo access
code-server    | setting sudo password using SUDO_PASSWORD env var
code-server    | Enter new UNIX password: Retype new UNIX password: passwd: password updated successfully
code-server    | [cont-init.d] 30-config: exited 0.
code-server    | [cont-init.d] 90-custom-folders: executing...
code-server    | [cont-init.d] 90-custom-folders: exited 0.
code-server    | [cont-init.d] 99-custom-scripts: executing...
code-server    | [custom-init] no custom files found exiting...
code-server    | [cont-init.d] 99-custom-scripts: exited 0.
code-server    | [cont-init.d] done.
code-server    | [services.d] starting services
code-server    | [services.d] done.
caddy_1        | {"level":"info","ts":1630062137.8387172,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
caddy_1        | {"level":"info","ts":1630062137.8424344,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["localhost:2019","[::1]:2019","127.0.0.1:2019"]}
caddy_1        | {"level":"info","ts":1630062137.8427486,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc000255c00"}
caddy_1        | {"level":"info","ts":1630062137.842788,"logger":"http","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
caddy_1        | {"level":"info","ts":1630062137.8428242,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
caddy_1        | {"level":"info","ts":1630062137.8435123,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/data/caddy"}
caddy_1        | {"level":"info","ts":1630062137.8435614,"logger":"tls","msg":"finished cleaning storage units"}
caddy_1        | {"level":"info","ts":1630062137.8435733,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["fabio.fontana.ns0.it"]}
caddy_1        | {"level":"info","ts":1630062137.8439515,"msg":"autosaved config (load with --resume flag)","file":"/config/caddy/autosave.json"}
caddy_1        | {"level":"info","ts":1630062137.8439713,"msg":"serving initial configuration"}
caddy_1        | {"level":"info","ts":1630062137.8444726,"logger":"tls.obtain","msg":"acquiring lock","identifier":"fabio.fontana.ns0.it"}
caddy_1        | {"level":"info","ts":1630062137.846244,"logger":"tls.obtain","msg":"lock acquired","identifier":"fabio.fontana.ns0.it"}
caddy_1        | {"level":"info","ts":1630062137.8717043,"logger":"tls.issuance.acme","msg":"waiting on internal rate limiter","identifiers":["fabio.fontana.ns0.it"],"ca":"https://acme.zerossl.com/v2/DV90","account":"fabio.fontana@mailbox.org"}
caddy_1        | {"level":"info","ts":1630062137.871737,"logger":"tls.issuance.acme","msg":"done waiting on internal rate limiter","identifiers":["fabio.fontana.ns0.it"],"ca":"https://acme.zerossl.com/v2/DV90","account":"fabio.fontana@mailbox.org"}
code-server    | [2021-08-27T11:02:19.498Z] info  code-server 3.11.1 acb5de66b71a3f8f1cc065df4633ecd3f9788d46
code-server    | [2021-08-27T11:02:19.501Z] info  Using user-data-dir ~/data
code-server    | [2021-08-27T11:02:19.530Z] info  Using config file ~/.config/code-server/config.yaml
code-server    | [2021-08-27T11:02:19.530Z] info  HTTP server listening on http://0.0.0.0:8443
code-server    | [2021-08-27T11:02:19.530Z] info    - Authentication is enabled
code-server    | [2021-08-27T11:02:19.531Z] info      - Using password from $PASSWORD
code-server    | [2021-08-27T11:02:19.531Z] info    - Not serving HTTPS
code-server    | [2021-08-27T11:02:19.531Z] info    - Proxying the following domain:
code-server    | [2021-08-27T11:02:19.531Z] info      - *.fabio-fontana.ns0.it
caddy_1        | {"level":"info","ts":1630062139.9989595,"logger":"tls.issuance.acme.acme_client","msg":"trying to solve challenge","identifier":"fabio.fontana.ns0.it","challenge_type":"http-01","ca":"https://acme.zerossl.com/v2/DV90"}

## Screenshot
![image](https://user-images.githubusercontent.com/38903806/131119481-abe5f744-f095-4a32-9574-8e06aea861a5.png)
![image](https://user-images.githubusercontent.com/38903806/131119527-72f1bd64-cb02-40df-81ea-089d97afe42f.png)
![image](https://user-images.githubusercontent.com/38903806/131119686-f1991fd3-42eb-4bb0-9471-76358475a7c0.png)
![image](https://user-images.githubusercontent.com/38903806/131119730-2595d3a7-51b0-45b7-8eed-ebb851cc50b0.png)

## Notes

This issue can be reproduced in VS Code: Yes

**Comment by jsjoeio** [comment]
Transferred to a discussion since I believe this could be related to your setup rather than code-server. 

```
caddy_1 | {"level":"info","ts":1630062139.9989595,"logger":"tls.issuance.acme.acme_client","msg":"trying to solve challenge","identifier":"fabio.fontana.ns0.it","challenge_type":"http-01","ca":"https://acme.zerossl.com/v2/DV90"}
```

Based on the prefix, it looks like this log is coming from Caddy.

How have you tried to troubleshoot this so far?

Not sure if this helps: [Caddy not responding to .well-known/acme-challenge](https://caddy.community/t/caddy-not-responding-to-well-known-acme-challenge/11588/4)
